server/src/main/java/org/apache/iotdb/db/mpp/execution/FragmentInstanceContext.java

```diff
@@ -69,4 +77,35 @@ public OperatorContext addOperatorContext(
   public List<OperatorContext> getOperatorContexts() {
     return operatorContexts;
   }
+
+  public FragmentInstanceId getId() {
+    return id;
+  }
+
+  public DriverContext getDriverContext() {
+    return driverContext;
+  }
+
+  public void setDriverContext(DriverContext driverContext) {
+    this.driverContext = driverContext;
+  }
+
+  public void failed(Throwable cause) {
+    state.set(FragmentInstanceState.FAILED);
+  }
+
+  public void cancel() {
+    state.set(FragmentInstanceState.CANCELED);
+  }
+
+  public void abort() {
+    state.set(FragmentInstanceState.ABORTED);
+  }
+
+  public void finish() {
+    if (state.get().isDone()) {
```

Comment:
Is there any other state execpt `FINISHED` when `state.get().isDone() == true`?

Reply:
<img width="1920" alt="image" src="https://user-images.githubusercontent.com/16079446/161372656-de727936-6a33-444c-9d1b-07778712a1ef.png">
These four states will all return true.